include/xtensor/xmath.hpp

```diff
@@ -231,6 +231,54 @@ namespace xt
         return detail::make_xfunction((functor_type)detail::clamp, std::forward<E1>(e1), std::forward<E2>(hi), std::forward<E3>(lo));
     }
 
+    namespace detail
+    {
+        template <typename T>
+        inline constexpr std::enable_if_t<std::is_signed<T>::value, T>
+        sign_impl(T x)
+        {
+            if (std::isnan(x))
+            {
+                return std::numeric_limits<T>::quiet_NaN();
+            }
+            return x == 0 ? copysign(T(0), x) : copysign(T(1), x);
```

Comment:
this is what causes the windows failure. One cannot have multiple `return` statements in a constexpr function. So this leaves you with

```
return std::isnan(x)? std::numeric_limits<T>::quiet_NaN() : (x == 0 ? copysign(T(0), x) : copysign(T(1), x));
```